lgc/patch/NggPrimShader.cpp

```diff
@@ -2592,9 +2592,20 @@ void NggPrimShader::distributePrimitiveId(Value *primitiveId) {
   {
     m_builder.SetInsertPoint(distribPrimitiveIdBlock);
 
-    Value *provokingVertexIndex = m_pipelineState->getRasterizerState().provokingVertexMode == ProvokingVertexFirst
-                                      ? m_nggInputs.vertexIndex0
-                                      : m_nggInputs.vertexIndex2;
+    auto primitiveType = m_pipelineState->getInputAssemblyState().primitiveType;
+    Value *provokingVertexIndex = nullptr;
+    if (primitiveType == lgc::PrimitiveType::Point) {
+      provokingVertexIndex = m_nggInputs.vertexIndex0;
+    } else if (primitiveType == lgc::PrimitiveType::LineList || primitiveType == lgc::PrimitiveType::LineStrip) {
+      provokingVertexIndex = m_pipelineState->getRasterizerState().provokingVertexMode == ProvokingVertexFirst
+                                 ? m_nggInputs.vertexIndex0
+                                 : m_nggInputs.vertexIndex1;
+    } else {
+      provokingVertexIndex = m_pipelineState->getRasterizerState().provokingVertexMode == ProvokingVertexFirst
```

Comment:
Here, I think we should make assert (primitiveType == TriangleXXX). This is because we still have Quad and Patch primitive type. For Quad, we don't support it. For Patch, it is not applicable here because this function is supposed to be called by pure VS without GS and tessellation.

Reply:
I think it should be `assert(primitiveType == TriangleXXX)` like Rex suggested.